src/AvlTree/AvlTreeNode.ts

```diff
@@ -14,7 +14,8 @@ class AvlTreeNode<ValueType> {
 
   public constructor(value: ValueType, index: number, comparator: Comparator) {
     this.value = value;
```

Comment:
Shouldn't we have an `index: number` property to keep the index of the current node?

Reply:
we have an `indexes: number[]` property, because the same value can be in two different positions in the Array.
Think in the following scenario:
Person 1) name: João Paulo, birthDay: 25/05/2005
Person 2) name: Maria Antonieta, birthDay: 25/05/2005

The tree of birthdays will have only one node for the date 25/05/2005, but we'll have two positions in the People Array, so the node for this date, should store both indexes.